src/Server/MySQLHandler.h

```diff
@@ -76,6 +88,10 @@ class MySQLHandler : public Poco::Net::TCPServerConnection
     using Replacements = std::unordered_map<std::string, ReplacementFn>;
     Replacements replacements;
 
+    std::mutex prepared_statements_mutex;
+    UInt32 current_prepared_statement_id = 0;
```

Comment:
Let's add `TSA_GUARDED_BY(prepared_statements_mutex)` behind l. 92 and l. 93. This enables Thread Safety Analysis (TSA) which is a compiler extension (kind of) that checks if synchronization is implemented properly.